dask_cloudprovider/providers/generic/vmcluster.py

```diff
@@ -0,0 +1,102 @@
+import asyncio
+import os
+
+from jinja2 import Environment, FileSystemLoader
+
+from distributed.deploy.spec import SpecCluster, ProcessInterface
+from distributed.utils import warn_on_duration
+
+
+class VM(ProcessInterface):
+    """A superclass for VM Schedulers, Workers and Nannies.
+
+    """
+
+    def __init__(self, **kwargs):
+        self.docker_image = "daskdev/dask:latest"
+        super().__init__(**kwargs)
+
+    def render_cloud_init(self, *args, **kwargs):
+        loader = FileSystemLoader([os.path.dirname(os.path.abspath(__file__))])
+        environment = Environment(loader=loader)
+        template = environment.get_template("cloud-init.yaml.j2")
+        return template.render(*args, **kwargs)
+
+    async def start(self):
+        """Create a VM."""
+        await super().start()
+
+    async def close(self):
+        """Destroy a VM."""
+        await super().close()
```

Comment:
These are here to be suggestive to downstream authors?

Reply:
Yeah.